xmake/rules/c++/modules/xmake.lua

```diff
@@ -21,45 +21,162 @@
 -- define rule: c++.build.modules
 rule("c++.build.modules")
     set_extensions(".mpp", ".mxx", ".cppm", ".ixx")
+    add_deps("c++.build.modules.dependencies")
+    add_deps("c++.build.modules.builder")
+    add_deps("c++.build.modules.install")
+
     on_config(function (target)
-        -- we disable to build across targets in parallel, because the source files may depend on other target modules
-        -- @see https://github.com/xmake-io/xmake/issues/1858
-        local target_with_modules
-        for _, dep in ipairs(target:orderdeps()) do
-            local sourcebatches = dep:sourcebatches()
-            if sourcebatches and sourcebatches["c++.build.modules"] then
-                target_with_modules = true
-                break
+        function contains_modules(target)
+            local target_with_modules
+            for _, dep in ipairs(target:orderdeps()) do
+                local sourcebatches = dep:sourcebatches()
+                if sourcebatches and sourcebatches["c++.build.modules"] then
+                    target_with_modules = true
+                    break
+                end
             end
+            return target_with_modules
         end
+
+        local target_with_modules = target:sourcebatches()["c++.build.modules"] and contains_modules(target) or false
```

Comment:
maybe we should rename `contains_modules` to `contains_modules_in_targetdeps`, because `target:set("policy", "build.across_targets_in_parallel", false)` only for it.

Reply:
so i should move target:set("policy", "build.across_targets_in_parallel", false) into a dedicated if ?